### PR TITLE
ReplyTo option for sharebymail Issue #24139

### LINKED ISSUE #24139: [sharebymail] Should let users decide if anwsers to be sent to their account email, or to nextcloud server email
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

**Is your feature request related to a problem? Please describe.**
Share by mail let user to share document with foreigner to get or send sensitive document protected by a password

Before mid-2019, the answers were sent to nextcloud server email. But since mid-2019, there were some modification of the code in order to change reply-to to the nextcloud account email (who has shared a document) and not to nextcloud server email anymore (new feature here #5637)

However, this situation is problematic because this lead to show private email to foreign people which is a lack of privacy. User should be able to decide if reply-to should be adressed to his email or to a generic email (nextcloud server email).

**Describe the solution you'd like**
The solution that could be applied is : in user parameters, to have a dedicated checkbox in order to let users to decide if reply-to is configured on their account email, or server one.

**Describe alternatives you've considered**
If not easy to introduce for each users, at least, to let admin to choose for all users the solution to apply in admin parameters section.

**Additional context**
Some people are using nextcloud because they want to keep their privacy and don't want people to know their personnal email. when sharing document or receiving documents (no personnal info that users cannot agree or not)

### PR DESCRIPTION
**Pull request by dems54**
Should let users decide if anwsers to be sent to their account email, or to nextcloud server email

fix #24139

### CODE REVIEW

**Comment by dems54** [comment]
For review please : @skjnldsv @rullzer @ChristophWurst @juliushaertl

(I hope this is the right way to ask for reviewers, this is my first PR, many thanks 🙂)

**Comment by skjnldsv** [comment]
Hey! Thanks :)
Can you rebase and squash your commits into one?

File: apps/sharebymail/lib/Settings/SettingsManager.php [hunk 1]
```diff
@@ -38,6 +38,8 @@ class SettingsManager {
 
 	private $enforcePasswordProtectionDefault = 'no';
 
+	private $replyToInitiator = 'yes';
```
Comment (juliushaertl) [inline apps/sharebymail/lib/Settings/SettingsManager.php:41]:
```suggestion
	private $replyToInitiatorDefault = 'yes';
```

File: apps/sharebymail/lib/Settings/SettingsManager.php [hunk 2]
```diff
@@ -61,4 +63,14 @@ public function enforcePasswordProtection(): bool {
...
 	}
+
+	/**
+	 * should add reply to with initiator mail
+	 *
+	 * @return bool
+	 */
+	public function replyToInitiator(): bool {
+		$replyToInitiator = $this->config->getAppValue('sharebymail', 'replyToInitiator', $this->replyToInitiator);
+		return $replyToInitiator === 'yes';
```
Comment (juliushaertl) [inline apps/sharebymail/lib/Settings/SettingsManager.php:74]:
```suggestion
		return $this->config->getAppValue('sharebymail', 'replyToInitiator', $this->replyToInitiatorDefault);
```

**Review by juliusknorr — APPROVED** [review]
Small nitpicks, otherwise fine by me :+1:

**Comment by dems54** [comment]
> Hey! Thanks :)
> Can you rebase and squash your commits into one?

Rebase and squash are OK. 
But I don't know why Static Code Analysis is now NOK ? It doesn't seem to relate to my commit...

**Comment by dems54** [comment]
> Small nitpicks, otherwise fine by me 👍

Good suggestions. 
This is my first PR : who must validte your propositions ?

**Comment by skjnldsv** [comment]
> But I don't know why Static Code Analysis is now NOK ? It doesn't seem to relate to my commit...

No, it doesn't indeed. Just ignore

**Review by skjnldsv — APPROVED** [review]

File: apps/sharebymail/lib/Settings/SettingsManager.php [hunk 3]
```diff
@@ -61,4 +63,13 @@ public function enforcePasswordProtection(): bool {
...
 		return $enforcePassword === 'yes';
 	}
+
+	/**
+	 * should add reply to with initiator mail
+	 *
+	 * @return bool
+	 */
+	public function replyToInitiator(): bool {
+		return $this->config->getAppValue('sharebymail', 'replyToInitiator', $this->replyToInitiatorDefault);
```
Comment (rullzer) [inline apps/sharebymail/lib/Settings/SettingsManager.php:73]: getAppValue returns a string. But you say bool. Either you need to compare here (and thus return a bool). Or change the function signature.

**Review by rullzer — CHANGES_REQUESTED** [review]

**Comment by dems54** [comment]
Possible to merge this PR in NC20 or NC21 ?

**Comment by dems54** [comment]
@rullzer it's ok for you ?

**Comment by skjnldsv** [comment]
NC22

**Pull request merged**